### PR TITLE
feat(core): adapt storage progress to payment system

### PR DESCRIPTION
**Pull request by JimmFly**
image

https://github.com/toeverything/AFFiNE/assets/102217452/b1ea7147-322a-4631-8888-65252d65fd1f

### CODE REVIEW

File: packages/frontend/core/src/components/affine/setting-modal/account-setting/index.tsx [hunk 1]
```diff
@@ -145,8 +152,18 @@ const StoragePanel = () => {
   const { data } = useQuery({
     query: allBlobSizesQuery,
   });
+  const [subscription] = useUserSubscription();
+  const plan = subscription?.plan ?? SubscriptionPlan.Free;
+  const maxLimit = plan === SubscriptionPlan.Free ? 10737418240 : 107374182400;
```
Comment (joooye34) [inline packages/frontend/core/src/components/affine/setting-modal/account-setting/index.tsx:157]: We can use tool to parse and format bytes https://github.com/visionmedia/bytes.js

File: packages/frontend/component/src/components/setting-components/share.css.ts [hunk 2]
```diff
@@ -115,31 +115,19 @@ globalStyle(`${storageProgressWrapper} .storage-progress-desc`, {
...
   borderRadius: '4px',
-  backgroundColor: 'var(--affine-pure-black-10)',
+  backgroundColor: 'var(--affine-black-10)',
   overflow: 'hidden',
 });
 export const storageProgressBar = style({
   height: '100%',
   backgroundColor: 'var(--affine-processing-color)',
   selectors: {
-    '&.warning': {
```
Comment (joooye34) [inline packages/frontend/component/src/components/setting-components/share.css.ts:125]: why this is deleted, I have seem this still exists in design file.

Comment (JimmFly) [reply packages/frontend/component/src/components/setting-components/share.css.ts:125]: I had a discussion with the designer @hwangdev97  about this colour and status, and this is the outcome of that discussion.

File: packages/frontend/core/src/components/affine/setting-modal/account-setting/index.tsx [hunk 3]
```diff
@@ -146,7 +155,22 @@ const StoragePanel = () => {
...
   });
 
-  const onUpgrade = useCallback(() => {}, []);
+  const [subscription] = useUserSubscription();
+  const plan = subscription?.plan ?? SubscriptionPlan.Free;
+  const maxLimit = useMemo(() => {
+    return plan === SubscriptionPlan.Free ? bytes('10GB') : bytes('100GB');
+  }, [plan]);
+
+  console.log('maxLimit', maxLimit);
```
Comment (joooye34) [inline packages/frontend/core/src/components/affine/setting-modal/account-setting/index.tsx:164]: remove it

File: packages/frontend/component/src/components/setting-components/storage-progess.tsx [hunk 4]
```diff
@@ -10,72 +11,68 @@ export interface StorageProgressProgress {
...
 }: StorageProgressProgress) => {
   const t = useAFFiNEI18N();
   const percent = useMemo(
     () => Math.round((value / upperLimit) * 100),
     [upperLimit, value]
   );
 
-  const used = useMemo(() => transformBytesToMB(value), [value]);
-  const max = useMemo(() => transformBytesToGB(upperLimit), [upperLimit]);
+  const used = useMemo(() => bytes(value), [value]);
```
Comment (joooye34) [inline packages/frontend/component/src/components/setting-components/storage-progess.tsx:29]: use `bytes.format`

File: packages/frontend/core/src/components/affine/setting-modal/account-setting/index.tsx [hunk 5]
```diff
@@ -146,7 +155,22 @@ const StoragePanel = () => {
     query: allBlobSizesQuery,
   });
 
-  const onUpgrade = useCallback(() => {}, []);
+  const [subscription] = useUserSubscription();
+  const plan = subscription?.plan ?? SubscriptionPlan.Free;
+  const maxLimit = useMemo(() => {
+    return plan === SubscriptionPlan.Free ? bytes('10GB') : bytes('100GB');
```
Comment (joooye34) [inline packages/frontend/core/src/components/affine/setting-modal/account-setting/index.tsx:161]: use `bytes.parse`

**Review by joooye34 — APPROVED** [review]

**Pull request merged**